tests/test_flow/test_persistence.py

```diff
@@ -1208,3 +1208,53 @@ def x_y():
     assert builder.build().get("x") == 1
     assert builder.build().get("y") == 2
     assert counter.times_called() == 1
+
+
+@pytest.mark.needs_parallel
+def test_entries_needed_in_memory(builder, make_counter):
+    """
+      ----> B ---->
+    E               A
+      ----> C ---->
+
+    A depends on B and C
```

Comment:
Our goal is to test an entry being moved from deferred to pending, but doesn't that depend on the order in which we process the entries? That is, we're assuming that D gets requested (but not in memory) by C before it gets requested (in memory) by B? Is it possible that D can be completed before it gets requested by C?